### ISSUE TITLE
Regarding code of DFR

### ISSUE THREAD

**Issue by beotborry** [BigML-CS-UCLA/SpuCo#20]
Hello, first of all, thank you for your wonderful summary of baselines on 'spurious correlation'.

I have a question about training code of 'DFR'.
In the original paper, D_{val} is splitted to D^1_{val/2}, D^2_{val/2} for hyper-parameter searching.
After searching hyper-parameter with D_{val/2}s, they utilize full D_{val}, not splitted, to train last linear layer for `n_retrain` times.

For example, number of each group in validation set of `Waterbirds` is [467, 466, 133, 133].

First, this dataset is splitted to [233, 233, 66, 66], [234, 233, 67, 67] for hyper-parameter searching.
And then, group-balanced version of full validation set [133, 133, 133, 133] is used for last layer retraining.

But, as far I understand, your code seems utilizing group-balanced D_{val/2} for last layer retraining, so I think there is an error on reproducing 'DFR'.

https://github.com/BigML-CS-UCLA/SpuCo/blob/63ee7c44f9dc7e06df864209a42272095b0856af/src/spuco/last_layer_retrain/dfr.py#L218

I think that the above line should be modified to `self.train_multiple_model(self.best_C, X_labeled, y_labeled, g_labeled)`

---
Update: add paragraph in the original paper

"For DFR_{Val}^{Tr} we split the validation in half, and use one half to tune the regularization strength parameter; then, we retrain the logistic regression with the optimal regularization on the full validation set."
---

I would appreciate if you check this issue and reply.
Thank you.

**Comment by YihaoXue** [comment]
Hello, thank you for bringing this to our attention! We have made the updates to the DFR code.
